frontend/src/lib/search-utils.ts

```diff
@@ -27,44 +27,38 @@ export const handleSemverOperatorComparison = (
   inputVersion: string,
   operator: SearchOperator
 ) => {
-  const coerceVersion = (version: string) => semver.coerce(version) ?? version
-  const coercedInputVersion = coerceVersion(inputVersion)
-  const coercedVersionToCompare = coerceVersion(versionToCompare)
-  const isValidSemver = (version: string | semver.SemVer) => semver.valid(version) !== null
```

Comment:
`coerceVersion` complicates things by mixing return types (string | semver.SemVer)
`isValidSemver` is not needed because null is falsy, and for the coerced values, we know they are valid